### PR TITLE
Implements Logging Strategy

### PR DESCRIPTION
**Pull request by dvonthenen**
Addresses the following issues: https://github.com/deepgram/deepgram-dotnet-sdk/issues/219

This implements the logging strategy to be used in this SDK. By default we will log informational messages so we can troubleshoot basic issues. Otherwise, you can enable more levels of debug to troubleshoot.

A side quest... 
- KeepAlives for LiveClient was implemented in this PR.
- `httpId` which just allows creating multiple clients.
- Since I had to go through every line of code to add trace statements, I used this opportunity to clean up a lot of the code. This is by no means complete and there is still an outstanding issue open to track this https://github.com/deepgram/deepgram-dotnet-sdk/issues/210

All examples updated to show how to use the logging functionality.

## Summary by CodeRabbit

- **New Features**
	- Enhanced logging across various components for improved debugging and information purposes.
	- Added `httpId` parameter to constructors for better tracking of HTTP requests.
	- Introduced logging functionality with different log levels.
	- Added licensing information and updated copyright notices.
- **Refactor**
	- Removed unnecessary logging and TODO comments.
	- Refactored methods for better error handling and cancellation logic.
	- Adjusted StreamCallbackResult logic in microphone operations.
- **Chores**
	- Updated namespace declarations and access modifiers.
	- Added and updated package references for logging capabilities.
- **Documentation**
	- Added summary comments for classes and methods.
	- Updated examples with licensing information and initialized logging options.

### CODE REVIEW

**Comment by dvonthenen** [comment]
@coderabbitai review

File: Deepgram/Abstractions/AbstractRestClient.cs [hunk 1]
```diff
@@ -401,20 +488,26 @@ internal AbstractRestClient(string? apiKey = null, DeepgramHttpClientOptions? op
...
             return result;
         }
         catch (Exception ex)
         {
-            Log.Exception(_logger, "PUT", ex);
+            Log.Error("PutAsync<S, T>", $"Exception: {ex}");
+            Log.Verbose("PutAsync<S, T>", "LEAVE");
             throw;
         }
     }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Abstractions/AbstractRestClient.cs:513]: Nice catch! Will fix!

File: Deepgram/Clients/Live/v1/Client.cs [hunk 2]
```diff
@@ -44,65 +52,94 @@ public Client(string? apiKey = null, DeepgramWsClientOptions? options = null)
...
             {
+                Log.Debug("Connect", $"header: {header.Key}={header.Value}");
                 _clientWebSocket.Options.SetRequestHeader(header.Key, header.Value);
             }
         }
         if (headers is not null)
         {
             foreach (var header in headers)
             {
+                Log.Debug("Connect", $"header: {header.Key}={header.Value}");
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:93]: Nice catch!

File: examples/streaming/microphone/Program.cs [hunk 3]
```diff
@@ -1,59 +1,118 @@
...
 namespace SampleApp
 {
     class Program
     {
         static async Task Main(string[] args)
         {
-            Library.Initialize();
+            // Initialize Library with default logging
+            // Normal logging is "Info" level
+            //Deepgram.Library.Initialize();
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/streaming/microphone/Program.cs:21]: This is just an example.

File: Deepgram/Clients/Live/v1/Client.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:93]: Good catch!

**Review by SandraRodgers — APPROVED** [review]

**Pull request merged**